### PR TITLE
Update create device location header

### LINKED ISSUE #5: Create Device Needs Location Header
Return the url for the created device resource in the location header

### LINKED ISSUE #7: Add Github Actions Pipeline
Need to add actions for build pipelines

### PR DESCRIPTION
**Pull request by Lyddon7522**
Closes #5 Closes #7 

Updated CreateDevice to include Location Haeder

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+name: MedTrackerAAPI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
```
Comment (benjaminsampica) [inline .github/workflows/dotnet.yml:16]: You’ll want the concurrency mechanism GitHub has too to prevent the same pipeline running overtop itself.

Comment (benjaminsampica) [reply .github/workflows/dotnet.yml:16]: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs

**Review by benjaminsampica — COMMENTED** [review]
Fixing the nullable issues is recommended, even in tests.

Near zero warnings/messages gang rise up.

**Review by benjaminsampica — APPROVED** [review]

File: src/MedTrackerAPI/Infrastructure/MedTrackerDBContext.cs [hunk 2]
```diff
@@ -20,11 +20,11 @@ public record Supply
...
 {
     public int Id { get; set; }
-    public required string Description { get; set; }
+    public required string? Description { get; set; }
     public string? SerialNumber { get; set; }
-    public required string  Manufacturer { get; set; }
-    public required string  Model { get; set; }
-    public required string  PartNumber { get; set; }
-    public required string  LotNumber { get; set; }
+    public required string?  Manufacturer { get; set; }
```
Comment (benjaminsampica) [inline src/MedTrackerAPI/Infrastructure/MedTrackerDBContext.cs:25]: FWIW required and nullable strings don’t have to go together

Comment (Lyddon7522) [reply src/MedTrackerAPI/Infrastructure/MedTrackerDBContext.cs:25]: It was the only way to get the warnings out of the compiler. I wonder if the NRT warnings could be tweaked though. I actually had it as a note to ask you about it. For the tests that is.

**Pull request merged**